### PR TITLE
fix steps panic

### PR DESCRIPTION
**Pull request by alixander**
just added the panicking test. i know it's not valid, but it shouldn't panic

### CODE REVIEW

**Review by nhooyr — APPROVED** [review]
oh yes definitely should not should be an error. I guess somewhere I'm assuming each child of a steps is a map.

**Comment by alixander** [comment]
> I guess somewhere I'm assuming each child of a steps is a map.

your hunch is correct

File: d2ir/compile.go [hunk 1]
```diff
@@ -66,6 +66,9 @@ func (c *compiler) compileSteps(m *Map) {
 		if i == 0 {
 			base = m.CopyBase(sf)
 		} else {
+			if steps.Fields[i-1].Map() == nil {
```
Comment (nhooyr) [inline d2ir/compile.go:69]: Check if there's a scalar/array too and if so, error.

**Review by nhooyr — CHANGES_REQUESTED** [review]

File: d2ir/compile.go (hunk 1 above)
Comment (nhooyr) [reply d2ir/compile.go:69]: None of the compiler functions return an error. You use `c.errorf()` to report it and skip the offending section.

**Review by nhooyr — CHANGES_REQUESTED** [review]

File: d2ir/compile.go (hunk 1 above)
Comment (nhooyr) [reply d2ir/compile.go:69]: my main point though is that a step *must* be either a map or not set to any value.

File: d2ir/compile.go [hunk 2]
```diff
@@ -59,13 +60,18 @@ func (c *compiler) compileSteps(m *Map) {
...
-		if sf.Map() == nil {
+		if sf.Map() == nil || sf.Primary() != nil {
+			c.errorf(sf.References[0].Context.Key, "invalid step")
 			continue
 		}
 		var base *Map
 		if i == 0 {
 			base = m.CopyBase(sf)
 		} else {
+			if steps.Fields[i-1].Map() == nil || steps.Fields[i-1].Primary() != nil {
```
Comment (nhooyr) [inline d2ir/compile.go:71]: Remove this and break above instead. This error isn't accurate anyway and reports the wrong step as being invalid.

**Review by nhooyr — APPROVED** [review]
✅

**Pull request merged**